Webpack Issues

My installed packages:

```
"devDependencies": {
  "babel-core": "^6.13.2",
  "babel-loader": "^6.2.5",
  "babel-plugin-transform-es2015-modules-commonjs": "^6.14.0",
  "babel-preset-babili": "0.0.1",
  "babili": "0.0.3",
  "browserify-versionify": "^1.0.6",
  "ify-loader": "^1.0.3",
  "shader-loader": "^1.2.1",
  "webpack": "^1.13.2",
  "webpack-dev-server": "^1.14.1"
}
```

My babel loader config:

```
{
  test: /\.js$/,
  exclude: /(node_modules)/,
  loader: `babel`,
  query: {
    presets: [`babili`],
    plugins: [`transform-es2015-modules-commonjs`]
  }
}
```

Error upon running `webpack-dev-server --inline --colors`:

```
ERROR in ./script.js
Module build failed: Error: Cannot find module 'babel-types'
    at Function.Module._resolveFilename (module.js:325:15)
    at Function.Module._load (module.js:276:25)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
    at Object.<anonymous> (/home/ubuntu/workspace/node_modules/babel-preset-babili/node_modules/babel-plugin-minify-mangle-names/lib/renamer.js:14:9)
    at Module._compile (module.js:409:26)
    at Object.Module._extensions..js (module.js:416:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
 @ multi main
```

I explicitly installed **babel-types**, and now get this error upon re-running:

```
ERROR in ./script.js
Module build failed: Error: Plugin 0 specified in "foreign" provided an invalid property of "name"
    at Plugin.init (/home/ubuntu/workspace/node_modules/babel-core/lib/transformation/plugin.js:138:13)
    at Function.normalisePlugin (/home/ubuntu/workspace/node_modules/babel-core/lib/transformation/file/options/option-manager.js:147:12)
    at /home/ubuntu/workspace/node_modules/babel-core/lib/transformation/file/options/option-manager.js:181:30
    at Array.map (native)
    at Function.normalisePlugins (/home/ubuntu/workspace/node_modules/babel-core/lib/transformation/file/options/option-manager.js:153:20)
    at OptionManager.mergeOptions (/home/ubuntu/workspace/node_modules/babel-core/lib/transformation/file/options/option-manager.js:245:36)
    at /home/ubuntu/workspace/node_modules/babel-core/lib/transformation/file/options/option-manager.js:289:14
    at /home/ubuntu/workspace/node_modules/babel-core/lib/transformation/file/options/option-manager.js:342:20
    at Array.map (native)
    at OptionManager.resolvePresets (/home/ubuntu/workspace/node_modules/babel-core/lib/transformation/file/options/option-manager.js:305:20)
 @ multi main
```

--- a/packages/babel-preset-babili/src/index.js
+++ b/packages/babel-preset-babili/src/index.js
@@ -2,10 +2,7 @@ const err = new Error("Babili requires babel-core>=6.14.0. Upgrade babel-core or
 
 function isVersion(versionStr) {
   const version = versionStr.split(".").map((p) => parseInt(p));
-  if (version[0] >= 6 && version[1] >= 14) {
-    return true;
-  }
-  return false;
+  return version[0] >= 6 && version[1] >= 14;
 }
 
 module.exports = {

--- a/packages/babel-preset-babili/src/index.js
+++ b/packages/babel-preset-babili/src/index.js
@@ -1,20 +1,53 @@
+const err = new Error("Babili is supported only for babel version >=6.14.0");
+
+function isVersion(versionStr) {
+  const version = versionStr.split(".").map((p) => parseInt(p));
+  if (version[0] >= 6 && version[1] >= 14) {
+    return true;
+  }
+  return false;
+}
+
 module.exports = {
-  minified: true,
-  plugins: [
-    require("babel-plugin-minify-constant-folding"),
-    require("babel-plugin-minify-dead-code-elimination"),
-    require("babel-plugin-minify-flip-comparisons"),
-    require("babel-plugin-minify-guarded-expressions"),
-    require("babel-plugin-minify-infinity"),
-    require("babel-plugin-minify-mangle-names"),
-    require("babel-plugin-minify-replace"),
-    require("babel-plugin-minify-simplify"),
-    require("babel-plugin-minify-type-constructors"),
-    require("babel-plugin-transform-member-expression-literals"),
-    require("babel-plugin-transform-merge-sibling-variables"),
-    require("babel-plugin-transform-minify-booleans"),
-    require("babel-plugin-transform-property-literals"),
-    require("babel-plugin-transform-simplify-comparison-operators"),
-    require("babel-plugin-transform-undefined-to-void"),
-  ],
+  plugins: [function({version}) {
+    if (!isVersion(version)) {
+      throw err;
+    }
+    return {
+      visitor: {}
+    };
+  }],
 };
+
+Object.defineProperty(module.exports, "buildPreset", {
+  configurable: true,
+  writable: true,
+  enumerable: false,
+  value: preset
+});
+
+function preset(context) {
+  if (!isVersion(context.version)) {
+    throw err;
+  }
+  return {
+    minified: true,
+    plugins: [
+      require("babel-plugin-minify-constant-folding"),
+      require("babel-plugin-minify-dead-code-elimination"),
+      require("babel-plugin-minify-flip-comparisons"),
+      require("babel-plugin-minify-guarded-expressions"),
+      require("babel-plugin-minify-infinity"),
+      require("babel-plugin-minify-mangle-names"),
+      require("babel-plugin-minify-replace"),
+      require("babel-plugin-minify-simplify"),
+      require("babel-plugin-minify-type-constructors"),
+      require("babel-plugin-transform-member-expression-literals"),
+      require("babel-plugin-transform-merge-sibling-variables"),
+      require("babel-plugin-transform-minify-booleans"),
+      require("babel-plugin-transform-property-literals"),
+      require("babel-plugin-transform-simplify-comparison-operators"),
+      require("babel-plugin-transform-undefined-to-void"),
+    ],
+  };
+}

--- a/packages/babel-preset-babili/src/index.js
+++ b/packages/babel-preset-babili/src/index.js
@@ -1,4 +1,4 @@
-const err = new Error("Babili is supported only for babel version >=6.14.0");
+const err = new Error("Babili requires babel-core>=6.14.0. Upgrade babel-core or tools dependent on babel-core to recent versions");
 
 function isVersion(versionStr) {
   const version = versionStr.split(".").map((p) => parseInt(p));

